### PR TITLE
Address lzstring decompress failure

### LINKED ISSUE #4555: [BUG]: corrupted full link URLs do not yield error messages
### Describe the bug

Loading a full link URL that is corrupted (e.g. truncated) should produce an error message, instead of reloading the last cached page. Truncation of such long URLs is an easily unnoticed side-effect of copy-pasting them in messaging apps. 

### Steps to reproduce

1. produce a full link URL from the share button on godbolt.org
2. truncate it, e.g. at 4096 characters
3. load the truncated address in a browser

### Expected behavior

The webpage should display an error page (not unlike the one you get when loaiding a wrong short link URL).

### Reproduction link

https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGIAKxmpK4AMngMmAByPgBGmMQgAOwAbAAcpAAOqAqETgwe3r4BQVk5jgJhEdEscQkp6XaYDnlCBEzEBAU%2BfoG2mPblDK3tBJVRsfFJabZtHV1FvQqzo%2BHjNZP1AJS2qF7EyOwcAPRHANSnAPJUpyyGJthmJqkaJgCcrwqnNMRLp/gKqDi%2BEYB1Ip0M6FOBAQmC%2BxC8hBMGgAgidLtdoUwCF8SDE8OgsAwocRMJgwQB3BDZWEsEhtWhQphLTBI1FnCzEXbABDYrBYhCncJEKEw07kki0dBgiHg2gM3bEMWoMmstEAdUIAto2U%2BqGuuEYYMcctOAlhwBJWPiNzuKLRACVMEsSLCvAppQxISTgExwiLYTFaHgFAoqF4GUosaRVWchOFgKcYUwAG6MWgATxubrJIqxpqJ0NhSmQJOxRAyMfRlxJMTBJFN1zjzDwYPw6F%2B4QUGTwJMrABUEHmFDCMjDiFKxVSvj3fm0vNyy6LkFSlESlJgpZWAJIF0VxYDhBjxxOocn%2BxOYFOMT4yy8dBCV%2B3U01ebF607LpjZEDVwWNvKJnQtCVgAIv0eALv6NIuh6kJxtqTBgBwnwxJyqAANa5tiVDauSG6VgAYky2IxJmhamsQTDIPQDanAA4hgJgAMwWKcW7%2BgwBzbgwyYYbC0KZr6kLCiwgnoKmgh7LsnwAkY0Z2mcA55uSmqnAwqA3Pi6A0ThEE8oKgiuJ8woAu0lbIjEqCpueyICHghg3LsggUggdCwoQpwZHsCheE6/auUYnxeMwBAEJgLAZGFkLppg2J%2BhknJKBRpzEAg6ZsAAdOZnofq5mA4WRVJeJcApdj2hBGoOvLZNiCX5fElYWPlLqygyYURfQxmil4CV6diSYdOCOVeSS8lsmxDBLPCDg3Dm/CKtCuynOh6l4UszGsYtxVUJyLDnt860KacarEZ5JKrgQsFiqpLB9ZmKmBS%2BBBdpa6CgVZmAALRBuhRZYr8qGoBgx4COeyanUwpEhtlkK3OgSEdrV3rMIIP5qlVBmzYKnzoO06HbnKPirC5ebBqcuEpb6SUyj5GRZB0G3%2BXmwrkTCfUCu%2B0Jkxal7Yu0wA%2BIwBDbjagkhvE2KxfEzAMglVn4iCKpHciOXaG6jhUIVsLkkw6a6tcDHCRpbAMFlR1CLr04/JdOKKgAEpeEmAVgnzqVzRhwqge0CacyZ4Jg5KVpE9bkVgmAZMSVFvtcDtymCCiWwd2KUnmojZpWACyH2QgqpwWjKGTtELnx%2Bpi/VMHTmZLFaY1okRvEkjnMeO4wpwAoB3IZkaGlUGI8pUFQhG7f6PanG4lpEGQQ3CZRk08yGgGfCpcqEfWZopU6BCUUKYLauO%2BvnuKxCSqccQ5C7zEEZW6M9zO2JKOgXgHMJMLexpZc4uGG%2B8bQwtMVfR0BzuQYFQaWYg27xAyELDaF5BTJyZC5Z8As3KVlaD2RupwerXWhKcRgyZBI5RJBJYAOYsAHDwKmN6gDRS0kLIqd8twGDoXCBOFOZY8BsEXmdBWlY7ZMEhKIX4rk9q7GwsPB2V4GAUlUn3LhK5hZHSuBvOIDJDAkNoO6U%2B90VywiEuCMqxB4z9g0kofqHC7DXD9MAPeglLLWVcp8KB8R3S8P4vCZkRtExXlwQARy8GIDc55aRCL0rXM4W5riilQHTakb0WLT2wQKVgeJyhunBBw8yPpOyLn4ggTkEQEm0mYPsNKtFDaPipoEjh3ZpJinaJjCRTsZQxCxAQGicsvBUPGhqHB4M/StPCpFDKpw7YcJsrQW6C5MzKFPA1I6ds9j0EhEmbSAgEw4VYMeHaXtzyxTHJeCM6F0zX1Uq5LAtT4T9JymgGo6sALoFPAwYxp9UChS9p1Rx8R8CDEqvEfiGl0B4RXkdHc4JcaMMcAUz8nYElQJllbJW3SpyP2tn8fKWtLi%2BDwFkMKggoQaXaCwBQZtxqRFWFCDhTpzxdiorkvMbAmR7CpbjAhkIjwLmebSDo4CTZglFCpHBDiTzEEMQIkkgcjohEtuDBgfjgwwgnJyOUx5S6ihoPETMAtaCDyOh4EBTksAvy5AKDhXL6Q/gsK%2BLhDyvaPidPEShgE36ORJGKSi0IYHzTUqec8W0OVHQsK8jm1xtRCIrt2CIC8ZQGCWMeQu4RjHEBNs9J1MDOTBWWbkzk5JPhCpafQMMDJMAxQUEzZOhBI0P0TPOXRA8gzNFBjKf4txjlHQzngVQgSBUcxiCiq03CsAJOZPw3uYawogHMm%2BTiujPi90VEyIa6jPi/SJKKP2AdS3UNhM28BZR/GDDqcyBJy9NzKz%2BMFEhoMqQirwIGHMjaTTnypemz05ldQFJ5nmMMxBmAHC%2BAYNgE4YpiJ2YtJKGy2CbtJRpIMC4wQxCte6scfw8aVt9sGW99BUEOsCQIDM%2BK/jBjQKmRUOQFwH3FIih0mADwCAPgCEVp5WyeHyl/Yc/CnQuT%2BZ5S8XFlaEM3h%2BByEQSN/CBbQPl0TYQietDcvy1DU5eypcKPuMCENljpEyxJojcGvvNsaBkewSGaJlEwacGrThgW8CQsEoCunqlUpgEjn0EM/uPD5bVexMGcTo4LdAE7zZPwFBkAwf6pY/vAWrWNf7ByQjqoXRuq8Fo6OUf0IM0k%2BXccLKPXqskKK41WPZ8JRJXxhUVLjYgK0nQmZytlxU9MD2mLfV8epJJaSOu2XtQ2CTYP6XfI0xgzXOvngOIZZLhi/0s26gQYJtUWMkrRBcAUVIzzBR%2Brk2EIXaWez2pSRW/pMygLlGAMA/kstLi9oXI8dHaJCt7nHU0JG9sTPwkdHpAocJA3bMe1yx8Hrfc5oMzeXhanfCo2cB2bcgZEn%2BAcH9NXIRArK8eUiXCn2VguAUiIqgzFEv6JYokWQ8JTxlNjuKRISyUrCacEIazwRUDK8tVaRI/TKACbGiIV1mCAcaqSWgAAvTACGYjDL7BpSkoHRTtGXJ9Rg4V0zU5Vi/fbfoBsU77oEi2bQpGJJeSfI%2B7ZGqWmYR7VmJBRVtzctOkUaT9yGAW%2BEjERVT6kiESyeJlq3zLZ9fd%2BgDATsnjPJ%2BIwr3xoEWHsIxBmNWbhszJaCZB9esKPGm4bUhZEeqS3oYBQo3CQkUMQuNDbZfiiOHKsTO6ZDH8K6sa0%2BBeeQ/gibAiXe0WCvn8QyegTABbg9OHII8UURTGpTTKNAwU7CaMwH48BVIoEwy8fz9okJVXmj3k4NJQtIGGKUNTgAUqgQruvOECFBN6s8t0lDEEzFVAU2gD8RCKxNXxHeUoIj8AZYUbDPIeS2hAzeQ8dkv82YFwPsNcdd2429lxqUt41lrQ5ZKFXFdchVoQM1JgEkDMOJJwNIqsJVxogFbY/1OYUt8BXVCVS1/5ThgNvVsIQ4dFqdg5FQv9yIZlcVsQAA1PABwesP0IVSiEhSsNwByfoJKcIXSBwZjNiDeOFN8RUYPGzfyWvOQqpIkURHWcccmdWBqeJD%2BGgTtWGFhefVybkQjBQArHXYbVcJKd8cINWDBNgQxS6MtXXciSZHkf2X4beKtThYUNAQyT0RXWrd%2BNVf2fYWEXw1ggCQGecfSMQNZbiBgQWSiO9Ag2Ed8AxQgG8JNUPNEJSXkbyXkYMdbLhEkCCHXGUThOqUBG9OSXhMmV8WbTyWZRgmRcSZxTbJo00GJB%2Bbic9PEWpFpUKGiUGciLbGFfrFuKRefYcIGTCZAvJM0YZdGZmIqYw8idVE%2BOIcmGqS%2BcyE0MmN2LY9abQ7qHzWVXyJImiZeCTKYggTpTMd8IhRgGzQUFgM1GfGuefMfPg6TJza0EyLwGIW6GQyg/2SUALcaJXScUmT4YQ6TV5AjTYj6fYJTHYhTbEaxEgTMGTUjQcE0IVck

### Screenshots

Not applicable

### Operating System

_No response_

### Browser version

_No response_

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
Fixes #4555. Might also address #4817.

### CODE REVIEW

File: static/url.ts [hunk 1]
```diff
@@ -89,20 +89,27 @@ export function deserialiseState(stateText: string): any {
     try {
         state = unrisonify(stateText);
         if (state && state.z) {
-            state = unrisonify(lzstring.decompressFromBase64(state.z));
+            const data = lzstring.decompressFromBase64(state.z);
+            // If lzstring fails to decompress this it'll return an empty string rather than throwing an error
+            if(data === '') {
```
Comment (mattgodbolt) [inline static/url.ts:94]: can you format please? And...I will FIX THE PRECOMMIT ARGH MY FAULT! :)

File: static/url.ts [hunk 2]
```diff
@@ -89,20 +89,27 @@ export function deserialiseState(stateText: string): any {
...
+                throw new Error('lzstring decompress error, url is corrupted');
+            }
+            state = unrisonify(data);
         }
     } catch (ex) {
         exception = ex;
     }
 
+    // This handles prehistoric urls, assumes rison fails with an error
```
Comment (mattgodbolt) [inline static/url.ts:103]: Thanks for the comment!

**Review by mattgodbolt — APPROVED** [review]

File: static/url.ts (hunk 1 above)
Comment (jeremy-rifkin) [reply static/url.ts:94]: Ah crap my bad, thank you for catching that!

**Comment by jeremy-rifkin** [comment]
This is live

**Pull request merged**